src/components/monitor/HeroSection.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from 'react';
+import Header from '@/components/shared/Header';
+import EmailToolbar from './components/EmailToolBar';
+import { HaveIbeenPwnedText } from '@/assets/types/have-i-been-pawned';
+import { PwnedSection } from './PwnedSection';
+import { AllGoodSection } from './AllGoodSection';
+
+interface HeroSectionProps {
+  textContent: HaveIbeenPwnedText['HeroSection'];
+}
+
+export const HeroSection: React.FC<HeroSectionProps> = ({ textContent }) => {
+  const [result, setResult] = useState<any[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [queryMade, setQueryMade] = useState(false);
+  const [showResult, setShowResult] = useState(false);
+
+  const handleResultChange = (data: any[]) => {
+    setQueryMade(true);
+    setResult(data);
+    setError(null);
+  };
+
+  const handleErrorChange = (err: string | null) => {
+    setQueryMade(true);
+    setResult([]);
+    setError(err);
+  };
+
+  useEffect(() => {
```

Comment:
Maybe you can wait for the result instead of managing it with an extra boolean. So we can remove this useless code.

Reply:
If I don't do it this way, it doesn't work properly.